auth/src/main/java/org/aerogear/auth/RoleKey.java

```diff
@@ -0,0 +1,66 @@
+package org.aerogear.auth;
+import java.util.Objects;
+
+/**
+ * Stores a user's keycloak roles information
+ */
+
+public class RoleKey {
+
+    private final String roleName;
+    private final RoleType roleType;
+    private final String clientID;
+
+    /**
+     * Creates a new RoleKey object
+     *
+     * @param role
+     * @param clientID
+     */
+    public RoleKey(final IRole role, final String clientID) {
+        roleName = role.getRoleName();
+        roleType = role.getRoleType();
+        this.clientID = clientID;
+    }
+
+    /**
+     * Creates a new RoleKey object
+     *
+     * @param roleName
+     * @param roleType
+     * @param clientId
+     */
+    public RoleKey(final String roleName, final String clientId, final RoleType roleType){
+        this.roleName = roleName;
```

Comment:
Add a check that roleName and roleType are not null, so that no other checks are needed later